### PR TITLE
Throw instead of exit

### PR DESCRIPTION
**Pull request by vemv**
* Implement `:throw-if-check-unsuccessful?` option
  * Fixes https://github.com/rm-hull/nvd-clojure/issues/50

### CODE REVIEW

**Comment by vemv** [comment]
Ready again

File: src/nvd/task/check.clj [hunk 1]
```diff
@@ -65,10 +65,16 @@
...
     project))
 
-(defn conditional-exit [project]
-  (if (:exit-after-check project)
-    (System/exit (if (:failed? project) -1 0))
-    project))
+(defn conditional-exit [{:keys [throw-if-check-unsuccessful? exit-after-check failed?]
+                         :as project}]
+  (cond
+    throw-if-check-unsuccessful?
```
Comment (rm-hull) [inline src/nvd/task/check.clj:71]: should `failed?` be incorporated into this condition otherwise it would always throw?

**Comment by vemv** [comment]
Good catch, went too quickly for it, will add test coverage 👍

**Comment by vemv** [comment]
Fixed, and added an integration test for it. All green now

Will add a commit with a release + actually perform it to batch review time

**Comment by vemv** [comment]
I shipped these changes via https://github.com/rm-hull/nvd-clojure/pull/94

**Pull request closed without merging**